labman/db/process.py

```diff
@@ -665,6 +1069,115 @@ def normalization_process(self):
+        dest_plate_name: str, optional
+            The name of the destination plate. Default: IndexPCRPlate
+
+        Returns
+        -------
+        str
+            The Echo formatted pick list
+        """
+        # check that arrays are the right size
+        if len(sample_names) != len(sample_wells) != len(indices):
+            raise ValueError(
+                'sample_names (%s) has a size different from sample_wells '
+                '(%s) or index list (%s)'
+                % (len(sample_names), len(sample_wells), len(indices)))
+
+        # header
+        picklist = [
+            'Sample\tSource Plate Name\tSource Plate Type\tSource Well\t'
+            'Transfer Volume\tIndex Name\tIndex Sequence\t'
+            'Destination Plate Name\tDestination Well']
+
+        # i5 additions
+        for i, (sample, well) in enumerate(zip(sample_names, sample_wells)):
+            picklist.append('\t'.join([
+                str(sample), indices.iloc[i]['i5 plate'], i5_plate_type,
+                indices.iloc[i]['i5 well'], str(i5_vol),
+                indices.iloc[i]['i5 name'], indices.iloc[i]['i5 sequence'],
+                dest_plate_name, well]))
+        # i7 additions
+        for i, (sample, well) in enumerate(zip(sample_names, sample_wells)):
+            picklist.append('\t'.join([
+                str(sample), indices.iloc[i]['i7 plate'], i7_plate_type,
+                indices.iloc[i]['i7 well'], str(i7_vol),
+                indices.iloc[i]['i7 name'], indices.iloc[i]['i7 sequence'],
+                dest_plate_name, well]))
+
+        return '\n'.join(picklist)
+
+    def genereate_echo_picklist(self):
+        """Generates Echo pick list for preparing the shotgun library
+
+        Returns
+        -------
+        str
+            The echo-formatted pick list
+        """
+        sample_names = []
+        sample_wells = []
+        indices = {'i5 name': {}, 'i5 plate': {}, 'i5 sequence': {},
+                   'i5 well': {}, 'i7 name': {}, 'i7 plate': {},
+                   'i7 sequence': {}, 'i7 well': {}, 'index combo': {},
+                   'index combo seq': {}}
+
+        for idx, well in enumerate(chain.from_iterable(self.plates[0].layout)):
+            # Add the sample well
+            sample_wells.append(well.well_id)
+            # Get the sample name - we need to go back to the SampleComposition
+            lib_comp = well.composition
+            sample_comp = lib_comp.normalized_gdna_composition\
+                .gdna_composition.sample_composition
+            sample_names.append(sample_comp.content)
+            # Retrieve all the information about the indices
+            i5_comp = lib_comp.i5_composition.primer_set_composition
+            i5_well = i5_comp.container
+            indices['i5 name'][idx] = i5_comp.external_id
+            indices['i5 plate'][idx] = i5_well.plate.external_id
+            indices['i5 sequence'][idx] = i5_comp.barcode
```

Comment:
maybe want to rename that property `i5_comp.sequence` to differentiate from the plate barcode?